src/server/scripts/EasternKingdoms/Stratholme/instance_stratholme.cpp

```diff
@@ -246,14 +259,41 @@ class instance_stratholme : public InstanceMapScript
                                 break;
                             case FAIL:
                                 DoRemoveAurasDueToSpellOnPlayers(SPELL_BARON_ULTIMATUM);
+                                if (Creature* ysida = instance->GetCreature(ysidaGUID))
+                                    ysida->CastSpell(ysida, SPELL_PERM_FEIGN_DEATH, true);
                                 EncounterState[0] = data;
                                 break;
                             case DONE:
                                 EncounterState[0] = data;
-                                if (Creature* ysidaTrigger = instance->GetCreature(ysidaTriggerGUID))
+
+                                if (Creature* ysida = instance->GetCreature(ysidaGUID))
                                 {
-                                    Position ysidaPos = ysidaTrigger->GetPosition();
-                                    ysidaTrigger->SummonCreature(NPC_YSIDA, ysidaPos, TEMPSUMMON_TIMED_DESPAWN, 1800000);
+                                    if (GameObject* cage = instance->GetGameObject(ysidaCageGUID))
+                                        cage->UseDoorOrButton();
+
+                                    float x, y, z;
+                                    //! This spell handles the Dead man's plea quest completion
+                                    ysida->CastSpell(nullptr, SPELL_YSIDA_SAVED, true);
+                                    ysida->SetWalk(true);
```

Comment:
@Riztazz why not removing the npc flag quest giver here and restore it on the DB to avoid:
```
Table creature_queststarter has creature entry (16031) for quest 8946, but npcflag does not include UNIT_NPC_FLAG_QUESTGIVER
Table creature_questender has creature entry (16031) for quest 8945, but npcflag does not include UNIT_NPC_FLAG_QUESTGIVER

Reply:
No, its wrong place to remove the flag, you should remove it in here https://github.com/TrinityCore/TrinityCore/pull/20749/files/9b312782acacb1f52ab48dcf128423716dad5366#diff-dcbd21640517a04c2516b641d913b949R171

and do UPDATE `creature_template` SET `npcflag` = `npcflag` |  2 WHERE `entry` = 16031;

could you commit that?

Reply:
Open a PR when you are back from xmas